### PR TITLE
ARM thumb has variable instruction sizes

### PR DESCRIPTION
**Pull request by thvdburgt**
### Description ###
Change `instruction_length` of ARM:thumb to `None` from 2, Thumb has no fixed size instructions.

### Motivation and Context ###
This fixes some problems with the context code for ARM thumb.

### How Has This Been Tested? ###
ARM

### Types of changes ###

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### Checklist ###

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read and agree to the **CONTRIBUTING** document.

### CODE REVIEW

File: gef.py [hunk 1]
```diff
@@ -6410,7 +6411,6 @@ def context_code(self):
         arch_name = arch.name().lower()
         if is_arm_thumb():
             arch_name += ":thumb"
-            pc   += 1
```
Comment (thvdburgt) [inline gef.py]: I assume this was meant as a fix for context code.

**Comment by hugsy** [comment]
I don't think that's true: 
>  Thumb instructions are each 16 bits long, and have a corresponding 32-bit ARM instruction that has the same effect on the processor model. 
Source: http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0210c/CACBCAAE.html

**Comment by thvdburgt** [comment]
Thumb was extended with Thumb-2, this is a backwards compatible superset of the original Thumb instruction set, so instructions can be either 2 or 4 bytes:
http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.ddi0290g/ch01s02s01.html

**Comment by hugsy** [comment]
Interesting, so if it's a different mode, it shouldn't replace the original mode, but simply be added. That would make it cleaner. @Grazfather opinion on this?

File: gef.py (hunk 1 above)
Comment (Grazfather) [reply gef.py]: Yeah, this is just to set the lsb to 1. Why do you want to remove it?

**Comment by thvdburgt** [comment]
I don't think it can be classified as a mode on itself. There is more information in this thread:
https://stackoverflow.com/a/28670713

**Comment by Grazfather** [comment]
I like any code that removes arch-specific (e.g. 'is x86').

File: gef.py (hunk 1 above)
Comment (thvdburgt) [reply gef.py]: It gives the wrong addresses in the code context, why do you need to set the lsb? I don't how it is used in `gef_disassemble`? Only seems incorrect to call `gdb_get_nth_previous_instruction_address` with something other than the correct `pc`. Can you explain that?

**Comment by Grazfather** [comment]
It's not a different mode: 32-bit and 16-bit thumb code can be intermixed (no need for bx or anything). You've found a bug, so thanks :)

**Comment by Grazfather** [comment]
Please squash

**Comment by hugsy** [comment]
+1 with @Grazfather , I'm all good for merging, just squash it all please.

**Pull request merged**